src/oiiotool/diff.cpp

```diff
@@ -175,63 +175,15 @@ int
 Oiiotool::do_action_flipdiff(ImageRecRef ir0, ImageRecRef ir1, Oiiotool& ot)
 {
     print("Computing diff of \"{}\" vs \"{}\"\n", ir0->name(), ir1->name());
-    read(ir0);
-    read(ir1);
-
-    int ret = DiffErrOK;
-    for (int subimage = 0; subimage < ir0->subimages(); ++subimage) {
-        if (subimage > 0 && !ot.allsubimages)
-            break;
-        if (subimage >= ir1->subimages())
-            break;
-
-        for (int m = 0; m < ir0->miplevels(subimage); ++m) {
-            if (m > 0 && !ot.allsubimages)
-                break;
-            if (m > 0 && ir0->miplevels(subimage) != ir1->miplevels(subimage)) {
-                print("Files do not match in their number of MIPmap levels\n");
-                ret = DiffErrDifferentSize;
-                break;
-            }
-
-            ImageBuf& img0((*ir0)(subimage, m));
-            ImageBuf& img1((*ir1)(subimage, m));
-            int npels = img0.spec().width * img0.spec().height
-                        * img0.spec().depth;
-            if (npels == 0)
-                npels = 1;  // Avoid divide by zero for 0x0 images
-            OIIO_ASSERT(img0.spec().format == TypeDesc::FLOAT);
-
-            // Compare the two images.
-            //
-            ImageBufAlgo::CompareResults cr;
+    
+           
+    ImageBufAlgo::CompareResults cr;
             
             
-            cr = ImageBufAlgo::compare_flip(img0, img1);
+    cr = ImageBufAlgo::compare_flip(ir0->name(), ir1->name());
```

Comment:
Are you assuming that any images you do a flip compare on exist as files on disk? Consider an oiiotool command like

    oiiotool fg.exr bg.exr -over ref.exr -flipdiff -o flip.exr

Here, something like that would mean that you're computing a composite iamge of "fg over bg" and comparing that to a reference file (but note, the thing you're comparing to could also be a "computed image").

It just has to be like all our other IBA functions: Taking two ImageBuf's, regardless of their original source.

I think you might be able to scavenge a bunch of the core algorithm code from those NVIDIA files rather from writing the entire thing from scratch, but throw away all the parts that deal with disk files and their own in-memory image representation, and retrofit it all to do its thing with ImageBuf. 

As an aside, I haven't really dug into the flip code myself or know a lot about the algorithm. I've seen "flip diff" images, so I know that one thing it can do is produce a kind of "heat map" of where the important differences are. But does it also give any meaningful single metric of image differences (like some kind of overall total error and/or worst-error-of-any-pixel)? If both are possible, I assume we would want both exposed.